biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -673,4 +679,15 @@ public URI makeDir(URI uri) throws URISyntaxException {
 		else
 			return uri;
 	}
+
+	public boolean isOffline() {
+		if (offline == null) {
+			return false;
+		}
+		return offline.get();
```

Comment:
There is a risk of NPE here if another thread calls setOffline(null) after the null check says not null. Better to copy into local var and then use that.

```java
AtomicBoolean o = offline;
if (o == null) {
	return false;
}
return o.get();
```